src/backend/linux_raw/fs/syscalls.rs

```diff
@@ -1090,6 +1090,60 @@ pub(crate) fn fcntl_lock(fd: BorrowedFd<'_>, operation: FlockOperation) -> io::R
     }
 }
 
+#[inline]
+pub(crate) fn fcntl_getlk(fd: BorrowedFd<'_>) -> io::Result<bool> {
+    use linux_raw_sys::general::F_WRLCK;
+    #[cfg(target_pointer_width = "64")]
+    use linux_raw_sys::general::{flock, F_GETLK};
+    #[cfg(target_pointer_width = "32")]
+    use linux_raw_sys::general::{flock64 as flock, F_GETLK64 as F_GETLK};
+
+    let mut lock = flock {
+        l_type: F_WRLCK as _,
+
+        // When `l_len` is zero, this locks all the bytes from
+        // `l_whence`/`l_start` to the end of the file, even as the
+        // file grows dynamically.
+        l_whence: SEEK_SET as _,
+        l_start: 0,
+        l_len: 0,
```

Comment:
It might be nice to give the user control over these fields via parameters to the function call.

Reply:
I agree that it'd be nice to expose this functionality, however the existing `fcntl_lock`  doesn't currently give the user control over these flags either, so this is ok for now.